### PR TITLE
added tls options

### PR DESCRIPTION
**Pull request by chiararelandini**
added tls options for https protocol to makeCall in serviceBuilder

### CODE REVIEW

File: lib/serviceBuilder.js [hunk 1]
```diff
@@ -67,6 +67,9 @@ function makeCallAsStream(endUrl, method, body, options) {
       headers: options.headers,
       timeout: options.timeout,
       agent: options.agent,
+      cert: options.cert,
```
Comment (davidebianchi) [inline lib/serviceBuilder.js:70]: Those options should be added also to `index.d.ts` file, to add to ts typings

File: tests/serviceBuilder.test.js [hunk 2]
```diff
@@ -1384,5 +1386,144 @@ tap.test('serviceBuilder', test => {
...
+          res.writeHead(401)
+          return res.end('{"status": "nok"}')
+        }
+
+        res.end('{"status": "ok"}')
+      })
+
+      const service = serviceBuilder('localhost', {}, {
+        protocol: 'https',
+        port: 3200,
```
Comment (davidebianchi) [inline tests/serviceBuilder.test.js:1428]: Also the service could add the tls options, right?

Can you add a test?

File: tests/serviceBuilder.test.js [hunk 3]
```diff
@@ -1384,5 +1386,178 @@ tap.test('serviceBuilder', test => {
...
+    assert.tearDown(() => {
+      nock.disableNetConnect()
+    })
+
+    async function createServer() {
+      return new Promise((resolve) => {
+        const server = https.createServer({
+          requestCert: true,
+          rejectUnauthorized: false,
+          ca: fs.readFileSync('tests/fixtures/keys/ca.crt'),
```
Comment (fredmaggiowski) [inline tests/serviceBuilder.test.js:1400]: can we disable these lint issues for these test cases? (or use the async version with await, if applicable)

**Review by fredmaggiowski — COMMENTED** [review]
The implementation is fine, I'd just try and fix the lint issues raised by GitHub on the test files (we could disable them at all for the test file).

Just one question: with this PR we provide the feature of creating a single serviceProxy with custom certificates/etc; could we implement this feature also on a library-wide scale? This way a user could configure the library once and then build all the desired services automatically inheriting the configuration. Maybe managed via some environment variable specificying where the files are located 🤔 ?

**Comment by chiararelandini** [comment]
@fredmaggiowski  I don't know if using the same settings for all services would work. Certificates and Certificate Authority might change as the contacted server changes. Or the kind of authorization might change, it is not necessarily true that all service proxies use https configurations if one does.

If you just refer to having an environment variable to save certificate and key files location, this can be certainly done. Would you like me to implement it?

**Comment by fredmaggiowski** [comment]
@chiararelandini I agree with your comment:

> Certificates and Certificate Authority might change as the contacted server changes.

this is true, especially if the lib is used in a microservice that contacts different external services, their certification authorities will often (if not always) differ.

however there could still be a case where my microservice will only contact internal services that share the same certification authorities and I'd like my lib to be configured once for all the instances of a serviceProxy I might create. I think that this would be a useful commodity, we could implement it later on but if you prefer.

**Comment by fredmaggiowski** [comment]
Speaking with @chiararelandini we resolved with a good solution that will be implemented in another PR, it is tracked in https://github.com/mia-platform/custom-plugin-lib/issues/85

**Review by fredmaggiowski — APPROVED** [review]

**Comment by fredmaggiowski** [comment]
LGTM @davidebianchi any comment?

**Review by davidebianchi — APPROVED** [review]

**Comment by davidebianchi** [comment]
LGTM

**Pull request merged**